packages/addon-dev/src/rollup-hbs-plugin.ts

```diff
@@ -19,5 +23,39 @@ export default function rollupHbsPlugin(): Plugin {
         id: id + '.js',
       };
     },
+
+    // template-only components may be imported
+    async resolveId(source, importer, options) {
+      if (!options.isEntry) {
+        let isLocal = source.startsWith('.');
+        let hasExt = path.extname(source);
+
+        // imports with extensions will resolve automatically
+        if (!isLocal || hasExt) return;
```

Comment:
I don't think `hasExt` is a good way to discriminate here. it's legal to say both `./components/foo` and `./component/foo.js`, and both should get the same handling (if the js doesn't exist and the hbs does, we synthesize the template-only component JS.)

Reply:
true true